### DISCUSSION TITLE
foreach?

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#7344] [Language Ideas]
I know it has been requested many times, but I wanted to ask for it again.  This would be super helpful for web scenarios where ```null``` is often returned instead of empty collections.

I appreciate that the team is solving big, hard problems (like collection literals) but I'd love it if a few of these minor papercut items were implemented too.

**Comment by FaustVX** [comment]
With the future collection literals, you can simply do
```C"
foreach (var item in collection ?? [])
{ }
```
[sharplab.io](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGED2AbbEDGMAlpgHYAyRMECU2yANDCAnKQwCYgDUAPgAIAGAAT8AjADpKpAI4BuALAAoZQDcoCYdiLIYwgLzCOEMFDjYYACiKkYAbQC6ASkUqlYTEij4AFsMvqmlQQALbCNlo6egD80cKOTsoA3gC+QA===)

**Reply by HaloFour** [reply]
It works, but it would be nice if the compiler could optimize away the allocation of the empty array and the loop when the collection is `null`.

**Comment by svick** [comment]
We don't want to have multiple discussions about the same suggestion, so I'm closing this as a duplicate of https://github.com/dotnet/csharplang/discussions/1081.
